src/depot/outdated.clj

```diff
@@ -0,0 +1,114 @@
+        system (maven/make-system)
+        session (maven/make-session system local-repo)
+        selected (:mvn/version coord)
+        artifact (maven/coord->artifact lib (assoc coord :mvn/version "[0,)"))
+        versions-req (doto (new VersionRangeRequest)
+                       (.setArtifact artifact)
+                       (.setRepositories remote-repos))
+        versions (->> (.resolveVersionRange system session versions-req)
+                      (.getVersions)
+                      (map str))]
+    {:selected selected
+     :types (group-by version-type versions)}))
+
+(defn find-latest [types consider-types]
+  (let [versions (->> (select-keys types consider-types)
+                      (vals)
+                      (apply concat))]
+    (-> (sort version/version-compare versions)
+        (last))))
+
+(defmulti -current-latest-map (fn [_ coord _] (ext/coord-type coord)))
+
+(defmethod -current-latest-map :default
+  [_ _ _]
+  nil)
+
+(defmethod -current-latest-map :mvn
+  [lib coord {:keys [deps-map consider-types]}]
+  (let [{:keys [types selected]} (coord->version-status lib coord deps-map)
+        latest                   (find-latest types consider-types)]
+    (when (and (not (str/blank? selected))
+               (not (str/blank? latest))
+               (= (version/version-compare latest selected) 1))
+      {"Current" selected
+       "Latest"  latest})))
+
+(defn- parse-git-ls-remote
+  "Returns a map of ref name to the latest sha for that ref name."
+  [s]
+  (let [lines (-> s
+                  (str/trim-newline)
+                  (str/split #"\n"))]
+    (into {}
+          (comp
+           (map (fn [x]
+                  (-> x
+                      (str/triml)
+                      (str/split #"\t")
+                      (reverse)
+                      (vec))))
+           (filter #(= 2 (count %))))
+          lines)))
+
+(defmethod -current-latest-map :git
+  [lib coord _]
+  (let [{:keys [exit out err]} (sh/sh "git" "ls-remote" (:git/url coord))
+        latest-remote-sha (get (parse-git-ls-remote out) "HEAD")]
+    (when (and (= exit 0)
+               (neg? (ext/compare-versions
+                      lib coord (assoc coord :sha latest-remote-sha) {})))
+      {"Current" (:sha coord)
+       "Latest"  latest-remote-sha})))
+
+(defn current-latest-map
+  "Returns a map containing `'Current'` and `'Latest'` if the dependency has a
+  newer version otherwise returns `nil`."
+  [lib coord data]
+  (-current-latest-map lib coord data))
+
+(defn gather-outdated [consider-types aliases]
+  (let [deps-map (-> (reader/clojure-env)
+                     (:config-files)
+                     (reader/read-deps))
+        args-map (deps/combine-aliases deps-map aliases)
+        overrides (:override-deps args-map)
+        all-deps (merge (:deps deps-map) (:extra-deps args-map))]
+    (->> (for [[lib coord] all-deps
+               :let [_ (prn `(current-latest-map ~lib
```

Comment:
Do you mean to prn here? It's pretty noisy in the output? If it's there to make the choices easier to understand I guess it makes more sense.

Reply:
Also worth noting that this kinda reintroduces an earlier problem where all of the maven output is interleaved with the actual useful information of what has been bumped. If no requests to maven are required then it's fine, I just see the clean version bump. All of the checks for newer versions just have side effects where they print out the "fetching ...." and URLs.

This was fixed before when the table output was added, so all of the changes are collected as the maven work is done and then printed out at the end. Maybe we could do something similar here? Or just log what versions changed and print them at the end instead of during the loop?